cluster/photon-controller/util.sh

```diff
@@ -560,11 +560,20 @@ function wait-minion-apis {
 #
 function setup-pod-routes {
     local node
+
     KUBE_NODE_BRIDGE_NETWORK=()
     for (( node=0; node<${#NODE_NAMES[@]}; node++)); do
+
+        # This happens in two steps (wait for an address, wait for a non 172.x.x.x address)
+        # because it's both simpler and more clear what's happening.
         try-until-success-ssh ${KUBE_NODE_IP_ADDRESSES[$node]} \
-            "Waiting for cbr0 bridge on ${NODE_NAMES[$node]}"  \
-            'sudo ifconfig cbr0 | grep -oP "inet addr:\K\S+"'
+            "Waiting for cbr0 bridge on ${NODE_NAMES[$node]} to have an address"  \
+            'sudo ifconfig cbr0  | grep -oP "inet addr:\K\S+"'
+
+        try-until-success-ssh ${KUBE_NODE_IP_ADDRESSES[$node]} \
+            "Waiting for cbr0 bridge on ${NODE_NAMES[$node]} to have correct address"  \
+            'sudo ifconfig cbr0  | grep -oP "inet addr:\K\S+" | grep -v  "^172."'
```

Comment:
It's a little bit hacky, but it works in almost all cases. I don't know what a perfect fix is. I'm thinking about it and will do something better in the future. 

The reason this almost always works is two-fold:
* The cbr0 bridge always comes up on a 172.x.x.x address, at least with how we're bringing it up. It's in the RFC 1918 private space for IP addresses.
* The normal configuration for Kubernetes is to not use the 172.x.x.x addresses for the final address.

So I agree with you, but this is a reasonable approach for now.